src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/service/LocationService.kt

```diff
@@ -77,6 +77,49 @@ class LocationService(
     return locationRepository.findOneByPrisonIdAndPathHierarchy(prisonId, code)?.toDto(includeChildren = includeChildren, includeHistory = includeHistory)
   }
 
+  fun getLocationsByKeys(keys: List<String>, includeChildren: Boolean = false, includeHistory: Boolean = false): List<LocationDTO> {
+    data class PrisonAndPathHierarchy(
+      val prisonId: String,
+      val pathHierarchy: String,
+    )
+    data class PrisonAndPathHierarchies(
+      val prisonId: String,
+      val pathHierarchy: List<String>,
+    )
+
+    val keysDeconstructedToPrisonIdsAndPaths = mutableListOf<PrisonAndPathHierarchy>()
+
+    keys.forEach {
+      if (it.contains("-")) {
+        val (prisonId, pathHierarchy) = it.split("-", limit = 2)
+        keysDeconstructedToPrisonIdsAndPaths += PrisonAndPathHierarchy(prisonId, pathHierarchy)
+      }
+    }
+
+    val prisonIds = keysDeconstructedToPrisonIdsAndPaths.map { it.prisonId }.distinct()
+    val locationsGroupedByPrisonId = mutableListOf<PrisonAndPathHierarchies>()
+    val pathHierarchies = mutableListOf<String>()
+
+    for (prisonId in prisonIds) {
+      for (location in keysDeconstructedToPrisonIdsAndPaths) {
+        if (location.prisonId == prisonId) {
+          pathHierarchies += location.pathHierarchy
+        }
+      }
+      locationsGroupedByPrisonId += PrisonAndPathHierarchies(prisonId, pathHierarchies)
+    }
+
+    val queryResponse = mutableListOf<Location>()
+    locationsGroupedByPrisonId.forEach { it ->
+      queryResponse += locationRepository.findAllByPrisonIdAndPathHierarchyIsIn(it.prisonId, it.pathHierarchy)
```

Comment:
This could be a bit slow as it will make individual call.  I think there are two options:
1. Just support locations from one prison at a time, (I can't think why they'd need to do it across prisons)
2. Write some JQL instead and do it one query e.g. concat two fields

Reply:
e.g. 
```sql
select l.* from location l
where concat(prison_id,'-',path_hierarchy) IN ('LEI-MW', 'MDI-1')
```

Reply:
@GurnankCheema 
Think this should do it in repo 
```kotlin
  @Query("select l from Location l where concat(l.prisonId,'-',l.pathHierarchy) IN (:keys)")
  fun findAllByKeys(keys: List<String>): List<Location>
```

Reply:
I will use the JQL as it gives us more flexibility for the future